// @mui-treasury/layout/components/Subheader/Subheader.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _clsx = _interopRequireDefault(require("clsx"));

var _useTheme = _interopRequireDefault(require("@material-ui/core/styles/useTheme"));

var _StyledProxy = require("../Shared/StyledProxy");

var _utils = require("../../utils");

var _contexts = require("../../contexts");

var _SubheaderCompiler = _interopRequireDefault(require("../../compilers/SubheaderCompiler"));

var _HeaderOffsetCompiler = _interopRequireDefault(require("../../compilers/HeaderOffsetCompiler"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var _default = function _default(styled) {
  var styledProxy = (0, _StyledProxy.generateStyledProxyCreator)(styled);
  var Div = styledProxy('div');

  var Subheader = function Subheader(_ref) {
    var className = _ref.className,
        subheaderId = _ref.subheaderId,
        props = _objectWithoutProperties(_ref, ["className", "subheaderId"]);

    if (!subheaderId) {
      throw new Error('You must provide "subheaderId" prop to <Subheader />');
    }

    var theme = (0, _useTheme["default"])();

    var _useLayoutCtx = (0, _contexts.useLayoutCtx)(),
        data = _useLayoutCtx.data,
        state = _useLayoutCtx.state;

    var compiler = (0, _SubheaderCompiler["default"])(state, data.subheader, data.edgeSidebar);
    var styles = (0, _utils.createBreakpointStyles)(compiler.getResultStyle(subheaderId), theme);
    var offsetCompiler = (0, _HeaderOffsetCompiler["default"])(data.subheader.configMapById[subheaderId]);
    var offsetStyles = (0, _utils.createBreakpointStyles)(offsetCompiler.getResultStyle(), theme);
    return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement(Div, _extends({
      className: (0, _clsx["default"])('Subheader', className),
      "mui-layout": subheaderId,
      styles: _extends({
        zIndex: 1000,
        transition: _utils.CSS_TRANSITION
      }, styles)
    }, props)), /*#__PURE__*/_react["default"].createElement(Div, {
      className: (0, _clsx["default"])('SubheaderOffset', className),
      styles: _extends({
        transition: _utils.CSS_TRANSITION,
        height: 0
      }, offsetStyles)
    }));
  };

  return Subheader;
};

exports["default"] = _default;
