// @mui-treasury/layout/components/InsetContainer/InsetContainer.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styles = require("@material-ui/core/styles");

var _Container = _interopRequireDefault(require("@material-ui/core/Container"));

var _FullscreenContext = require("../../contexts/FullscreenContext");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n    overflow: auto;\n    flex-grow: 1;\n  "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var _default = function _default(styled) {
  var Div = styled('div')(_templateObject());
  var useContainerStyles = (0, _styles.makeStyles)({
    root: {
      display: 'flex',
      flexFlow: 'row nowrap',
      flexGrow: 1
    }
  });

  var InsetContainer = function InsetContainer(_ref) {
    var children = _ref.children,
        leftSidebar = _ref.leftSidebar,
        rightSidebar = _ref.rightSidebar,
        WrapperProps = _ref.WrapperProps,
        props = _objectWithoutProperties(_ref, ["children", "leftSidebar", "rightSidebar", "WrapperProps"]);

    var isFullscreen = (0, _FullscreenContext.useFullscreenCtx)();
    var classes = useContainerStyles(props);
    return /*#__PURE__*/_react["default"].createElement(_Container["default"], _extends({}, props, {
      classes: classes
    }), leftSidebar, isFullscreen ? /*#__PURE__*/_react["default"].createElement(Div, WrapperProps, children) : children, rightSidebar);
  };

  return InsetContainer;
};

exports["default"] = _default;
