// @mui-treasury/layout/models/Sidebar/Inset/FixedInset.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var _default = function _default(config) {
  var isHidden = config && config.hidden;
  return {
    getRootStyle: function getRootStyle() {
      if (isHidden) return {
        display: 'none'
      };
      return {
        width: config.width
      };
    },
    getBodyStyle: function getBodyStyle() {
      if (isHidden) return {
        display: 'none'
      };
      return _extends(_extends({
        position: 'fixed',
        top: 0,
        width: 'auto',
        height: '100%'
      }, config.anchor === 'left' && {
        marginLeft: -999,
        paddingLeft: 999
      }), config.anchor === 'right' && {
        marginRight: -999,
        paddingRight: 999
      });
    }
  };
};

exports["default"] = _default;
