// @mui-treasury/layout/components/Shared/SharedSidebar.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.createSwipeableVariant = exports.createDrawerVariant = exports.StyledProxySwipeableDrawer = exports.StyledProxyDrawer = exports.CLS = void 0;

var _react = _interopRequireDefault(require("react"));

var _Drawer = _interopRequireDefault(require("@material-ui/core/Drawer"));

var _SwipeableDrawer = _interopRequireDefault(require("@material-ui/core/SwipeableDrawer"));

var _utils = require("../../utils");

var _StyledProxy = _interopRequireDefault(require("./StyledProxy"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var CLS = 'MuiTreasury-paper';
exports.CLS = CLS;
var StyledProxyDrawer = (0, _StyledProxy["default"])(_Drawer["default"], CLS);
exports.StyledProxyDrawer = StyledProxyDrawer;
var StyledProxySwipeableDrawer = (0, _StyledProxy["default"])(_SwipeableDrawer["default"], CLS);
exports.StyledProxySwipeableDrawer = StyledProxySwipeableDrawer;

var createDrawerVariant = function createDrawerVariant(variant) {
  var StyledComponent = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : StyledProxyDrawer;

  var Sidebar = function Sidebar(_ref) {
    var _ref$PaperProps = _ref.PaperProps,
        PaperProps = _ref$PaperProps === void 0 ? {} : _ref$PaperProps,
        props = _objectWithoutProperties(_ref, ["PaperProps"]);

    return /*#__PURE__*/_react["default"].createElement(StyledComponent, _extends({}, props, {
      PaperProps: _extends(_extends({}, PaperProps), {}, {
        className: "".concat(CLS, " ").concat(PaperProps.className)
      }),
      variant: variant
    }));
  };

  Sidebar.displayName = "".concat((0, _utils.upperFirst)(variant), "DrawerSidebar");
  return Sidebar;
};

exports.createDrawerVariant = createDrawerVariant;

var createSwipeableVariant = function createSwipeableVariant(variant) {
  var StyledComponent = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : StyledProxySwipeableDrawer;

  var Sidebar = function Sidebar(_ref2) {
    var _ref2$PaperProps = _ref2.PaperProps,
        PaperProps = _ref2$PaperProps === void 0 ? {} : _ref2$PaperProps,
        props = _objectWithoutProperties(_ref2, ["PaperProps"]);

    return /*#__PURE__*/_react["default"].createElement(StyledComponent, _extends({}, props, {
      PaperProps: _extends(_extends({}, PaperProps), {}, {
        className: "".concat(CLS, " ").concat(PaperProps.className)
      }),
      variant: variant
    }));
  };

  Sidebar.displayName = "".concat((0, _utils.upperFirst)(variant), "SwipeableSidebar");
  return Sidebar;
};

exports.createSwipeableVariant = createSwipeableVariant;
