// @mui-treasury/layout/components/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Root", {
  enumerable: true,
  get: function get() {
    return _Root["default"];
  }
});
Object.defineProperty(exports, "getInsetContainer", {
  enumerable: true,
  get: function get() {
    return _InsetContainer["default"];
  }
});
Object.defineProperty(exports, "getFullscreen", {
  enumerable: true,
  get: function get() {
    return _Fullscreen["default"];
  }
});
Object.defineProperty(exports, "getSidebarContent", {
  enumerable: true,
  get: function get() {
    return _SidebarContent["default"];
  }
});
Object.defineProperty(exports, "getHeader", {
  enumerable: true,
  get: function get() {
    return _Header["default"];
  }
});
Object.defineProperty(exports, "getSubheader", {
  enumerable: true,
  get: function get() {
    return _Subheader["default"];
  }
});
Object.defineProperty(exports, "getContent", {
  enumerable: true,
  get: function get() {
    return _Content["default"];
  }
});
Object.defineProperty(exports, "getCollapseIcon", {
  enumerable: true,
  get: function get() {
    return _CollapseIcon["default"];
  }
});
Object.defineProperty(exports, "getCollapseBtn", {
  enumerable: true,
  get: function get() {
    return _CollapseBtn["default"];
  }
});
Object.defineProperty(exports, "getSidebarTrigger", {
  enumerable: true,
  get: function get() {
    return _SidebarTrigger["default"];
  }
});
Object.defineProperty(exports, "getDrawerSidebar", {
  enumerable: true,
  get: function get() {
    return _DrawerSidebar["default"];
  }
});
Object.defineProperty(exports, "getFooter", {
  enumerable: true,
  get: function get() {
    return _Footer["default"];
  }
});
Object.defineProperty(exports, "getInsetAvoidingView", {
  enumerable: true,
  get: function get() {
    return _InsetAvoidingView["default"];
  }
});
Object.defineProperty(exports, "getInsetFooter", {
  enumerable: true,
  get: function get() {
    return _InsetFooter["default"];
  }
});
Object.defineProperty(exports, "getInsetSidebar", {
  enumerable: true,
  get: function get() {
    return _InsetSidebar["default"];
  }
});
Object.defineProperty(exports, "getSwipeableSidebar", {
  enumerable: true,
  get: function get() {
    return _SwipeableSidebar["default"];
  }
});

var _Root = _interopRequireDefault(require("./Root"));

var _InsetContainer = _interopRequireDefault(require("./InsetContainer"));

var _Fullscreen = _interopRequireDefault(require("./Fullscreen"));

var _SidebarContent = _interopRequireDefault(require("./SidebarContent"));

var _Header = _interopRequireDefault(require("./Header"));

var _Subheader = _interopRequireDefault(require("./Subheader"));

var _Content = _interopRequireDefault(require("./Content"));

var _CollapseIcon = _interopRequireDefault(require("./CollapseIcon"));

var _CollapseBtn = _interopRequireDefault(require("./CollapseBtn"));

var _SidebarTrigger = _interopRequireDefault(require("./SidebarTrigger"));

var _DrawerSidebar = _interopRequireDefault(require("./DrawerSidebar"));

var _Footer = _interopRequireDefault(require("./Footer"));

var _InsetAvoidingView = _interopRequireDefault(require("./InsetAvoidingView"));

var _InsetFooter = _interopRequireDefault(require("./InsetFooter"));

var _InsetSidebar = _interopRequireDefault(require("./InsetSidebar"));

var _SwipeableSidebar = _interopRequireDefault(require("./SwipeableSidebar"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
