// @mui-treasury/layout/esm/effects/MultiHeaders/MultiHeadersEffect.js
import HeaderEffect from '../../effects/Header/HeaderEffect';
import MultiHeadersModel from '../../models/MultiHeaders';
import { subtractCalc, plusCalc } from '../../utils';
export default (function (configs) {
  return {
    getSidebarInteraction: function getSidebarInteraction(sidebarId) {
      var clippedConfigs = configs.filter(function (c) {
        return HeaderEffect(c).isObjectClipped(sidebarId);
      });
      var relativeConfigs = MultiHeadersModel(clippedConfigs).relativeConfigs;
      var relativeHeight = MultiHeadersModel(relativeConfigs).totalHeight;
      var nonRelativeConfigs = MultiHeadersModel(clippedConfigs).nonRelativeConfigs;
      var nonRelativeHeight = MultiHeadersModel(nonRelativeConfigs).totalHeight;
      var maxHeightFromTop = MultiHeadersModel(nonRelativeConfigs).maxHeightFromTop;
      var nonRelativeOffset = subtractCalc(nonRelativeHeight, maxHeightFromTop);
      var maxOffset = plusCalc(relativeHeight, nonRelativeOffset);
      return {
        isClipped: clippedConfigs.length > 0,
        clippedHeight: MultiHeadersModel(clippedConfigs).totalHeight,
        relativeHeight: relativeHeight,
        nonRelativeHeight: nonRelativeHeight,
        maxHeightFromTop: maxHeightFromTop,
        nonRelativeOffset: nonRelativeOffset,
        maxOffset: maxOffset
      };
    }
  };
});
