// @mui-treasury/layout/esm/hooks/useSidebar.js
import { useTheme } from '@material-ui/core/styles';
import EdgeSidebarCompiler from '../compilers/EdgeSidebarCompiler';
import { useLayoutCtx } from '../contexts';
import { getSidebarAnchor, createDisplayNone } from '../utils';
export var useSidebar = function useSidebar(id, consumer) {
  if (!id) {
    throw new Error("You must specify a sidebar id to <".concat(consumer, " />"));
  }

  var _useTheme = useTheme(),
      breakpoints = _useTheme.breakpoints;

  var _useLayoutCtx = useLayoutCtx(),
      state = _useLayoutCtx.state,
      data = _useLayoutCtx.data,
      setOpen = _useLayoutCtx.setOpen,
      setCollapsed = _useLayoutCtx.setCollapsed;

  var sidebarConfigMap = data.edgeSidebar.configMapById[id];
  var anchor = getSidebarAnchor(sidebarConfigMap);
  var styles = EdgeSidebarCompiler(state, data.edgeSidebar, data.header).getResultStyle(id);
  return {
    anchor: anchor,
    sidebarConfigMap: sidebarConfigMap,
    state: state.sidebar[id],
    edgeSidebar: data.edgeSidebar,
    hiddenStyles: createDisplayNone(data.edgeSidebar.hiddenById[id], breakpoints),
    styles: styles,
    setOpen: setOpen,
    setCollapsed: setCollapsed
  };
};
export default useSidebar;
