// @mui-treasury/layout/components/CollapseBtn/CollapseBtn.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styles = require("@material-ui/core/styles");

var _Button = _interopRequireDefault(require("@material-ui/core/Button"));

var _KeyboardArrowLeftRounded = _interopRequireDefault(require("@material-ui/icons/KeyboardArrowLeftRounded"));

var _KeyboardArrowRightRounded = _interopRequireDefault(require("@material-ui/icons/KeyboardArrowRightRounded"));

var _StyledProxy = require("../Shared/StyledProxy");

var _hooks = require("../../hooks");

var _resolveDirection = require("../../utils/resolveDirection");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var _default = function _default(styled) {
  var styledProxy = (0, _StyledProxy.generateStyledProxyCreator)(styled);
  var StyledButton = styledProxy(_Button["default"]);
  var useStyles = (0, _styles.makeStyles)(function (_ref) {
    var palette = _ref.palette,
        breakpoints = _ref.breakpoints;
    return {
      root: _defineProperty({
        backgroundColor: palette.type === 'dark' ? palette.background.paper : palette.grey[50],
        textAlign: 'center',
        borderRadius: 0,
        borderTop: '1px solid',
        borderColor: palette.action.hover
      }, breakpoints.up('sm'), {
        minHeight: 40
      })
    };
  });

  var CollapseBtn = function CollapseBtn(_ref2) {
    var children = _ref2.children,
        sidebarId = _ref2.sidebarId,
        _onClick = _ref2.onClick,
        SvgIconProps = _ref2.SvgIconProps,
        props = _objectWithoutProperties(_ref2, ["children", "sidebarId", "onClick", "SvgIconProps"]);

    var classes = useStyles(props);

    var _useSidebarCollapse = (0, _hooks.useSidebarCollapse)(sidebarId, 'CollapseBtn'),
        id = _useSidebarCollapse.id,
        anchor = _useSidebarCollapse.anchor,
        hiddenStyles = _useSidebarCollapse.hiddenStyles,
        state = _useSidebarCollapse.state,
        setCollapsed = _useSidebarCollapse.setCollapsed;

    var theme = (0, _styles.useTheme)();

    var arrowR = /*#__PURE__*/_react["default"].createElement(_KeyboardArrowRightRounded["default"], SvgIconProps);

    var arrowL = /*#__PURE__*/_react["default"].createElement(_KeyboardArrowLeftRounded["default"], SvgIconProps);

    var getArrow = function getArrow(direction) {
      if (direction === 'left') {
        return state.collapsed ? arrowR : arrowL;
      }

      if (direction === 'right') {
        return state.collapsed ? arrowL : arrowR;
      }

      return null;
    };

    return /*#__PURE__*/_react["default"].createElement(StyledButton, _extends({}, props, {
      classes: classes,
      styles: hiddenStyles,
      onClick: function onClick(e) {
        if (typeof _onClick === 'function') _onClick(e);
        setCollapsed(id, !state.collapsed);
      }
    }), typeof children === 'function' ? children(_extends({
      anchor: anchor
    }, state)) : getArrow(theme.direction === 'rtl' ? (0, _resolveDirection.switchDirection)(anchor) : anchor));
  };

  return CollapseBtn;
};

exports["default"] = _default;
