// @mui-treasury/layout/esm/models/Header/HeaderModel.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import { isHeaderClipped } from '../../utils';
import createEffect from '../../effects/Header/HeaderEffect';
import createWidthInterface from '../Width';
import createMarginInterface from '../Margin';
export default (function (config, sidebarEffects) {
  var headerEffect = createEffect(config);
  var marginInterfaces = [];
  var widthInterfaces = [];
  sidebarEffects.forEach(function (_ref) {
    var sidebarId = _ref.id,
        getObjectWidth = _ref.getObjectWidth,
        getObjectMargin = _ref.getObjectMargin;
    widthInterfaces.push(isHeaderClipped(config, sidebarId) ? createWidthInterface(0) : getObjectWidth(config.id));
    marginInterfaces.push(isHeaderClipped(config, sidebarId) ? createMarginInterface({
      marginLeft: 0,
      marginRight: 0
    }) : getObjectMargin(config.id));
  });
  var marginStyle = marginInterfaces.length > 0 ? marginInterfaces.reduce(function (result, current) {
    return result.combine(current);
  }).getStyle() : undefined;
  var widthStyle = widthInterfaces.length > 0 ? widthInterfaces.reduce(function (result, current) {
    return result.combine(current);
  }).getStyle() : undefined;
  return {
    getStyle: function getStyle() {
      return _extends(_extends(_extends(_extends({
        height: config.hidden ? 0 : config.initialHeight,
        position: config.position
      }, config.top !== undefined && {
        top: config.top
      }), marginStyle), widthStyle), headerEffect.getHeaderZIndex());
    }
  };
});
