// @mui-treasury/layout/shared/State.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.createMultiObjData = exports.createSingleObjData = void 0;

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var createSingleObjData = function createSingleObjData(_ref) {
  var id = _ref.id,
      _ref$props = _ref.props,
      props = _ref$props === void 0 ? {} : _ref$props,
      _ref$rpsConfig = _ref.rpsConfig,
      rpsConfig = _ref$rpsConfig === void 0 ? {} : _ref$rpsConfig,
      _ref$hidden = _ref.hidden,
      hidden = _ref$hidden === void 0 ? [] : _ref$hidden;
  return {
    id: id,
    props: props,
    rpsConfig: rpsConfig,
    hidden: hidden
  };
};

exports.createSingleObjData = createSingleObjData;

var createMultiObjData = function createMultiObjData(rpsConfigById) {
  if (!rpsConfigById) return {
    ids: [],
    dataById: {}
  };
  var ids = Object.keys(rpsConfigById);
  return {
    ids: ids,
    dataById: ids.reduce(function (result, id) {
      return _extends(_extends({}, result), {}, _defineProperty({}, id, createSingleObjData({
        id: id,
        rpsConfig: rpsConfigById[id]
      })));
    }, {})
  };
};

exports.createMultiObjData = createMultiObjData;
