// @mui-treasury/layout/effects/PersistentSidebar/PersistentSidebarEffect.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _utils = require("../../utils");

var _Edge = _interopRequireDefault(require("../../models/Sidebar/Edge"));

var _Width = _interopRequireDefault(require("../../models/Width"));

var _Margin = _interopRequireDefault(require("../../models/Margin"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

var _default = function _default(config, state) {
  var anchor = config.anchor;

  var _createEdgeModel = (0, _Edge["default"])(config, state),
      currentWidth = _createEdgeModel.width;

  var marginAttr = "margin".concat((0, _utils.upperFirst)(anchor));

  var isBehavior = function isBehavior(value, objectId) {
    if (_typeof(config.persistentBehavior) === 'object') {
      var objectBehavior = (0, _utils.get)(config, ['persistentBehavior', objectId]);

      if (!objectBehavior) {
        return (0, _utils.get)(config, 'persistentBehavior._other') === value;
      }

      return objectBehavior === value;
    }

    return config.persistentBehavior === value;
  };

  var isSidebarOpen = (0, _utils.get)(state, ['sidebar', config.id, 'open']);
  return {
    id: config.id,
    getObjectWidth: function getObjectWidth(objectId) {
      return (0, _Width["default"])(isSidebarOpen && isBehavior('fit', objectId) ? currentWidth : 0);
    },
    getObjectMargin: function getObjectMargin(objectId) {
      var getResult = function getResult() {
        if (!isSidebarOpen) {
          return _defineProperty({}, marginAttr, 0);
        } // open is true


        if (isBehavior('fit', objectId)) {
          return _defineProperty({}, marginAttr, currentWidth);
        }

        if (isBehavior('flexible', objectId)) {
          // for flexible, only marginLeft works. You can try using marginRight.
          // May be it is because normally browser is LTR
          return {
            marginLeft: (0, _utils.getFlexBehaviorValue)(config.anchor, currentWidth)
          };
        }

        return _defineProperty({}, marginAttr, 0);
      };

      return (0, _Margin["default"])(getResult());
    }
  };
};

exports["default"] = _default;
