// @mui-treasury/layout/hooks/useSidebar.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.useSidebar = void 0;

var _styles = require("@material-ui/core/styles");

var _EdgeSidebarCompiler = _interopRequireDefault(require("../compilers/EdgeSidebarCompiler"));

var _contexts = require("../contexts");

var _utils = require("../utils");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var useSidebar = function useSidebar(id, consumer) {
  if (!id) {
    throw new Error("You must specify a sidebar id to <".concat(consumer, " />"));
  }

  var _useTheme = (0, _styles.useTheme)(),
      breakpoints = _useTheme.breakpoints;

  var _useLayoutCtx = (0, _contexts.useLayoutCtx)(),
      state = _useLayoutCtx.state,
      data = _useLayoutCtx.data,
      setOpen = _useLayoutCtx.setOpen,
      setCollapsed = _useLayoutCtx.setCollapsed;

  var sidebarConfigMap = data.edgeSidebar.configMapById[id];
  var anchor = (0, _utils.getSidebarAnchor)(sidebarConfigMap);
  var styles = (0, _EdgeSidebarCompiler["default"])(state, data.edgeSidebar, data.header).getResultStyle(id);
  return {
    anchor: anchor,
    sidebarConfigMap: sidebarConfigMap,
    state: state.sidebar[id],
    edgeSidebar: data.edgeSidebar,
    hiddenStyles: (0, _utils.createDisplayNone)(data.edgeSidebar.hiddenById[id], breakpoints),
    styles: styles,
    setOpen: setOpen,
    setCollapsed: setCollapsed
  };
};

exports.useSidebar = useSidebar;
var _default = useSidebar;
exports["default"] = _default;
