// @mui-treasury/layout/models/Width/WidthModel.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.sumExternalGap = exports.getCssWidth = void 0;

var _utils = require("../../utils");

var getCssWidth = function getCssWidth(externalGap) {
  if (typeof externalGap === "string") {
    return "calc(100% - (".concat(externalGap, "))");
  }

  if (typeof externalGap === "number") {
    if (externalGap === 0) return "100%";
    return "calc(100% - ".concat(externalGap, "px)");
  }

  return "auto";
};

exports.getCssWidth = getCssWidth;

var sumExternalGap = function sumExternalGap(a, b) {
  if ((0, _utils.isNilOrEmpty)(a) && (0, _utils.isNilOrEmpty)(b)) return undefined;
  if (a === 0 && b === 0) return 0;
  if (!a && b) return b;
  if (a && !b) return a;
  return "".concat((0, _utils.toValidCssValue)(a), " + ").concat((0, _utils.toValidCssValue)(b));
};

exports.sumExternalGap = sumExternalGap;

var createWidthInterface = function createWidthInterface(externalGap) {
  return {
    value: externalGap,
    combine: function combine(w) {
      return createWidthInterface(sumExternalGap(externalGap, w.value));
    },
    getStyle: function getStyle() {
      return {
        width: getCssWidth(externalGap)
      };
    }
  };
};

var _default = createWidthInterface;
exports["default"] = _default;
