// @mui-treasury/layout/esm/components/InsetSidebar/InsetSidebar.js
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { generateStyledProxyCreator } from '../Shared/StyledProxy';
import getInsetHeaderOffset from '../InsetHeaderOffset';
import { useInsetSidebar } from '../../hooks';
export default (function (styled) {
  var styledProxy = generateStyledProxyCreator(styled);
  var Div = styledProxy('div');
  var InsetHeaderOffset = getInsetHeaderOffset(styled);
  var useStyles = makeStyles(function (_ref) {
    var palette = _ref.palette;
    return {
      root: {
        position: 'relative',
        flexShrink: 0
      },
      paper: {
        backgroundColor: palette.type === 'dark' ? palette.background.paper : palette.grey[100]
      }
    };
  });

  var InsetSidebar = function InsetSidebar(_ref2) {
    var sidebarId = _ref2.sidebarId,
        children = _ref2.children,
        props = _objectWithoutProperties(_ref2, ["sidebarId", "children"]);

    var classes = useStyles(props);

    var _useInsetSidebar = useInsetSidebar(sidebarId),
        rootStyles = _useInsetSidebar.rootStyles,
        bodyStyles = _useInsetSidebar.bodyStyles;

    return /*#__PURE__*/React.createElement(Div, {
      className: "InsetSidebar-root ".concat(classes.root),
      styles: rootStyles
    }, /*#__PURE__*/React.createElement(Div, {
      className: "InsetSidebar-paper ".concat(classes.paper),
      styles: bodyStyles
    }, /*#__PURE__*/React.createElement(InsetHeaderOffset, {
      sidebarId: sidebarId
    }), children));
  };

  return InsetSidebar;
});
