// @mui-treasury/layout/esm/utils/sidebarChecker.js
export var hasAutoExpanded = function hasAutoExpanded(config) {
  return !!config && typeof config.autoExpanded === 'boolean';
};
export var isCollapsibleSidebarConfig = function isCollapsibleSidebarConfig(config) {
  return !!config && typeof config.collapsible === "boolean";
};
export var isPersistentSidebarConfig = function isPersistentSidebarConfig(config) {
  return !!config && config.variant === "persistent";
};
export var isPermanentSidebarConfig = function isPermanentSidebarConfig(config) {
  return !!config && config.variant === "permanent";
};
export var isTemporarySidebarConfig = function isTemporarySidebarConfig(config) {
  return !!config && config.variant === "temporary";
};
export var isFixedInsetSidebarConfig = function isFixedInsetSidebarConfig(config) {
  return !!config && config.variant === "fixed";
};
export var isAbsoluteInsetSidebarConfig = function isAbsoluteInsetSidebarConfig(config) {
  return !!config && config.variant === "absolute";
};
export var isStickyInsetSidebarConfig = function isStickyInsetSidebarConfig(config) {
  return !!config && config.variant === "sticky";
};
