// @mui-treasury/layout/utils/sidebarChecker.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.isStickyInsetSidebarConfig = exports.isAbsoluteInsetSidebarConfig = exports.isFixedInsetSidebarConfig = exports.isTemporarySidebarConfig = exports.isPermanentSidebarConfig = exports.isPersistentSidebarConfig = exports.isCollapsibleSidebarConfig = exports.hasAutoExpanded = void 0;

var hasAutoExpanded = function hasAutoExpanded(config) {
  return !!config && typeof config.autoExpanded === 'boolean';
};

exports.hasAutoExpanded = hasAutoExpanded;

var isCollapsibleSidebarConfig = function isCollapsibleSidebarConfig(config) {
  return !!config && typeof config.collapsible === "boolean";
};

exports.isCollapsibleSidebarConfig = isCollapsibleSidebarConfig;

var isPersistentSidebarConfig = function isPersistentSidebarConfig(config) {
  return !!config && config.variant === "persistent";
};

exports.isPersistentSidebarConfig = isPersistentSidebarConfig;

var isPermanentSidebarConfig = function isPermanentSidebarConfig(config) {
  return !!config && config.variant === "permanent";
};

exports.isPermanentSidebarConfig = isPermanentSidebarConfig;

var isTemporarySidebarConfig = function isTemporarySidebarConfig(config) {
  return !!config && config.variant === "temporary";
};

exports.isTemporarySidebarConfig = isTemporarySidebarConfig;

var isFixedInsetSidebarConfig = function isFixedInsetSidebarConfig(config) {
  return !!config && config.variant === "fixed";
};

exports.isFixedInsetSidebarConfig = isFixedInsetSidebarConfig;

var isAbsoluteInsetSidebarConfig = function isAbsoluteInsetSidebarConfig(config) {
  return !!config && config.variant === "absolute";
};

exports.isAbsoluteInsetSidebarConfig = isAbsoluteInsetSidebarConfig;

var isStickyInsetSidebarConfig = function isStickyInsetSidebarConfig(config) {
  return !!config && config.variant === "sticky";
};

exports.isStickyInsetSidebarConfig = isStickyInsetSidebarConfig;
