// @mui-treasury/layout/esm/EdgeSidebar/EdgeSidebarRegistry.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var EdgeSidebarRegistry = function EdgeSidebarRegistry(state, props) {
  return {
    registerPermanentConfig: function registerPermanentConfig(breakpoint, config) {
      state.rpsConfig[breakpoint] = _extends(_extends(_extends({}, config), props), {}, {
        id: state.id,
        variant: 'permanent'
      });
      return this;
    },
    registerPersistentConfig: function registerPersistentConfig(breakpoint, config) {
      state.rpsConfig[breakpoint] = _extends(_extends(_extends({}, config), props), {}, {
        id: state.id,
        variant: 'persistent'
      });
      return this;
    },
    registerTemporaryConfig: function registerTemporaryConfig(breakpoint, config) {
      state.rpsConfig[breakpoint] = _extends(_extends(_extends({}, config), props), {}, {
        id: state.id,
        variant: 'temporary'
      });
      return this;
    }
  };
};

export default EdgeSidebarRegistry;
