// @mui-treasury/layout/esm/models/Width/WidthModel.js
import { isNilOrEmpty, toValidCssValue } from "../../utils";
export var getCssWidth = function getCssWidth(externalGap) {
  if (typeof externalGap === "string") {
    return "calc(100% - (".concat(externalGap, "))");
  }

  if (typeof externalGap === "number") {
    if (externalGap === 0) return "100%";
    return "calc(100% - ".concat(externalGap, "px)");
  }

  return "auto";
};
export var sumExternalGap = function sumExternalGap(a, b) {
  if (isNilOrEmpty(a) && isNilOrEmpty(b)) return undefined;
  if (a === 0 && b === 0) return 0;
  if (!a && b) return b;
  if (a && !b) return a;
  return "".concat(toValidCssValue(a), " + ").concat(toValidCssValue(b));
};

var createWidthInterface = function createWidthInterface(externalGap) {
  return {
    value: externalGap,
    combine: function combine(w) {
      return createWidthInterface(sumExternalGap(externalGap, w.value));
    },
    getStyle: function getStyle() {
      return {
        width: getCssWidth(externalGap)
      };
    }
  };
};

export default createWidthInterface;
