// @mui-treasury/layout/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  LayoutConsumer: true,
  useLayoutCtx: true,
  WindowConsumer: true,
  WindowProvider: true,
  useWindowCtx: true,
  SidebarConsumer: true,
  useSidebarCtx: true,
  useScreen: true,
  useScrollY: true,
  useHeaderHeight: true,
  useSidebarTrigger: true,
  useSidebarCollapse: true
};
Object.defineProperty(exports, "default", {
  enumerable: true,
  get: function get() {
    return _builders["default"];
  }
});
Object.defineProperty(exports, "LayoutConsumer", {
  enumerable: true,
  get: function get() {
    return _contexts.LayoutConsumer;
  }
});
Object.defineProperty(exports, "useLayoutCtx", {
  enumerable: true,
  get: function get() {
    return _contexts.useLayoutCtx;
  }
});
Object.defineProperty(exports, "WindowConsumer", {
  enumerable: true,
  get: function get() {
    return _contexts.WindowConsumer;
  }
});
Object.defineProperty(exports, "WindowProvider", {
  enumerable: true,
  get: function get() {
    return _contexts.WindowProvider;
  }
});
Object.defineProperty(exports, "useWindowCtx", {
  enumerable: true,
  get: function get() {
    return _contexts.useWindowCtx;
  }
});
Object.defineProperty(exports, "SidebarConsumer", {
  enumerable: true,
  get: function get() {
    return _contexts.SidebarConsumer;
  }
});
Object.defineProperty(exports, "useSidebarCtx", {
  enumerable: true,
  get: function get() {
    return _contexts.useSidebarCtx;
  }
});
Object.defineProperty(exports, "useScreen", {
  enumerable: true,
  get: function get() {
    return _hooks.useScreen;
  }
});
Object.defineProperty(exports, "useScrollY", {
  enumerable: true,
  get: function get() {
    return _hooks.useScrollY;
  }
});
Object.defineProperty(exports, "useHeaderHeight", {
  enumerable: true,
  get: function get() {
    return _hooks.useHeaderHeight;
  }
});
Object.defineProperty(exports, "useSidebarTrigger", {
  enumerable: true,
  get: function get() {
    return _hooks.useSidebarTrigger;
  }
});
Object.defineProperty(exports, "useSidebarCollapse", {
  enumerable: true,
  get: function get() {
    return _hooks.useSidebarCollapse;
  }
});

var _builders = _interopRequireDefault(require("./builders"));

var _presets = require("./presets");

Object.keys(_presets).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _presets[key];
    }
  });
});

var _components = require("./components");

Object.keys(_components).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _components[key];
    }
  });
});

var _contexts = require("./contexts");

var _hooks = require("./hooks");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
