// @mui-treasury/layout/effects/MultiHeaders/MultiHeadersEffect.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _HeaderEffect = _interopRequireDefault(require("../../effects/Header/HeaderEffect"));

var _MultiHeaders = _interopRequireDefault(require("../../models/MultiHeaders"));

var _utils = require("../../utils");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var _default = function _default(configs) {
  return {
    getSidebarInteraction: function getSidebarInteraction(sidebarId) {
      var clippedConfigs = configs.filter(function (c) {
        return (0, _HeaderEffect["default"])(c).isObjectClipped(sidebarId);
      });
      var relativeConfigs = (0, _MultiHeaders["default"])(clippedConfigs).relativeConfigs;
      var relativeHeight = (0, _MultiHeaders["default"])(relativeConfigs).totalHeight;
      var nonRelativeConfigs = (0, _MultiHeaders["default"])(clippedConfigs).nonRelativeConfigs;
      var nonRelativeHeight = (0, _MultiHeaders["default"])(nonRelativeConfigs).totalHeight;
      var maxHeightFromTop = (0, _MultiHeaders["default"])(nonRelativeConfigs).maxHeightFromTop;
      var nonRelativeOffset = (0, _utils.subtractCalc)(nonRelativeHeight, maxHeightFromTop);
      var maxOffset = (0, _utils.plusCalc)(relativeHeight, nonRelativeOffset);
      return {
        isClipped: clippedConfigs.length > 0,
        clippedHeight: (0, _MultiHeaders["default"])(clippedConfigs).totalHeight,
        relativeHeight: relativeHeight,
        nonRelativeHeight: nonRelativeHeight,
        maxHeightFromTop: maxHeightFromTop,
        nonRelativeOffset: nonRelativeOffset,
        maxOffset: maxOffset
      };
    }
  };
};

exports["default"] = _default;
