// @mui-treasury/layout/esm/components/Subheader/Subheader.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import cx from 'clsx';
import useTheme from '@material-ui/core/styles/useTheme';
import { generateStyledProxyCreator } from '../Shared/StyledProxy';
import { createBreakpointStyles, CSS_TRANSITION } from '../../utils';
import { useLayoutCtx } from '../../contexts';
import SubheaderCompiler from '../../compilers/SubheaderCompiler';
import HeaderOffsetCompiler from '../../compilers/HeaderOffsetCompiler';
export default (function (styled) {
  var styledProxy = generateStyledProxyCreator(styled);
  var Div = styledProxy('div');

  var Subheader = function Subheader(_ref) {
    var className = _ref.className,
        subheaderId = _ref.subheaderId,
        props = _objectWithoutProperties(_ref, ["className", "subheaderId"]);

    if (!subheaderId) {
      throw new Error('You must provide "subheaderId" prop to <Subheader />');
    }

    var theme = useTheme();

    var _useLayoutCtx = useLayoutCtx(),
        data = _useLayoutCtx.data,
        state = _useLayoutCtx.state;

    var compiler = SubheaderCompiler(state, data.subheader, data.edgeSidebar);
    var styles = createBreakpointStyles(compiler.getResultStyle(subheaderId), theme);
    var offsetCompiler = HeaderOffsetCompiler(data.subheader.configMapById[subheaderId]);
    var offsetStyles = createBreakpointStyles(offsetCompiler.getResultStyle(), theme);
    return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Div, _extends({
      className: cx('Subheader', className),
      "mui-layout": subheaderId,
      styles: _extends({
        zIndex: 1000,
        transition: CSS_TRANSITION
      }, styles)
    }, props)), /*#__PURE__*/React.createElement(Div, {
      className: cx('SubheaderOffset', className),
      styles: _extends({
        transition: CSS_TRANSITION,
        height: 0
      }, offsetStyles)
    }));
  };

  return Subheader;
});
