// @mui-treasury/layout/esm/InsetSidebar/InsetSidebarRegistry.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var InsetSidebarRegistry = function InsetSidebarRegistry(state, props) {
  return {
    registerFixedConfig: function registerFixedConfig(breakpoint, config) {
      state.rpsConfig[breakpoint] = _extends(_extends(_extends({}, config), props), {}, {
        id: state.id,
        variant: 'fixed'
      });
      return this;
    },
    registerAbsoluteConfig: function registerAbsoluteConfig(breakpoint, config) {
      state.rpsConfig[breakpoint] = _extends(_extends(_extends({}, config), props), {}, {
        id: state.id,
        variant: 'absolute'
      });
      return this;
    },
    registerStickyConfig: function registerStickyConfig(breakpoint, config) {
      state.rpsConfig[breakpoint] = _extends(_extends(_extends({}, config), props), {}, {
        id: state.id,
        variant: 'sticky'
      });
      return this;
    }
  };
};

export default InsetSidebarRegistry;
