// @mui-treasury/layout/presets/cozy.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _builders = _interopRequireDefault(require("../builders"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var _default = function _default() {
  var scheme = (0, _builders["default"])();
  scheme.configureHeader(function (builder) {
    builder.registerConfig("xs", {
      position: "fixed",
      initialHeight: 56
    }).registerConfig("sm", {
      position: "fixed",
      initialHeight: 64
    }).registerConfig("md", {
      position: "fixed",
      clipped: true,
      initialHeight: 64
    });
  });
  scheme.configureEdgeSidebar(function (builder) {
    builder.create("primarySidebar", {
      anchor: "left"
    }).registerTemporaryConfig("xs", {
      width: 256
    }).registerPermanentConfig("sm", {
      width: 64,
      collapsible: false
    }).registerPermanentConfig("md", {
      width: 256,
      collapsible: true,
      collapsedWidth: 64
    });
  });
  scheme.enableAutoCollapse("primarySidebar", "md");
  return scheme;
};

exports["default"] = _default;
