// @mui-treasury/layout/presets/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "getContentBasedScheme", {
  enumerable: true,
  get: function get() {
    return _contentBased["default"];
  }
});
Object.defineProperty(exports, "getCozyScheme", {
  enumerable: true,
  get: function get() {
    return _cozy["default"];
  }
});
Object.defineProperty(exports, "getDefaultScheme", {
  enumerable: true,
  get: function get() {
    return _default["default"];
  }
});
Object.defineProperty(exports, "getFixedScheme", {
  enumerable: true,
  get: function get() {
    return _fixed["default"];
  }
});
Object.defineProperty(exports, "getMuiTreasuryScheme", {
  enumerable: true,
  get: function get() {
    return _muiTreasury["default"];
  }
});
Object.defineProperty(exports, "getStandardScheme", {
  enumerable: true,
  get: function get() {
    return _standard["default"];
  }
});

var _contentBased = _interopRequireDefault(require("./contentBased"));

var _cozy = _interopRequireDefault(require("./cozy"));

var _default = _interopRequireDefault(require("./default"));

var _fixed = _interopRequireDefault(require("./fixed"));

var _muiTreasury = _interopRequireDefault(require("./muiTreasury"));

var _standard = _interopRequireDefault(require("./standard"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
